ui/core/components/inputs/consumables.ts

```diff
@@ -0,0 +1,297 @@
+
+export const FOOD_CONFIG = [
+	{ config: HotWolfRibs, stats: [Stat.StatSpirit] },
+	{ config: SmokedSagefish, stats: [Stat.StatMP5] },
+	{ config: NightfinSoup, stats: [Stat.StatMP5, Stat.StatSpellPower] },
+	{ config: GrilledSquid, stats: [Stat.StatAgility] },
+	{ config: SmokedDesertDumpling, stats: [Stat.StatStrength] },
+	{ config: RunnTumTuberSurprise, stats: [Stat.StatIntellect] },
+	{ config: DirgesKickChimaerokChops, stats: [Stat.StatStamina] },
+	{ config: BlessSunfruit, stats: [Stat.StatStrength] },
+	{ config: BlessedSunfruitJuice, stats: [Stat.StatSpirit] },
+] as ConsumableStatOption<Food>[];
+
+export const makeFoodInput = makeConsumeInputFactory({consumesFieldName: 'food'});
+
+///////////////////////////////////////////////////////////////////////////
+//                                 PHYSICAL DAMAGE CONSUMES
+///////////////////////////////////////////////////////////////////////////
+
+// Agility
+export const ElixirOfTheMongoose = { actionId: ActionId.fromItemId(13452), value: AgilityElixir.ElixirOfTheMongoose, minLevel: 46 };
+export const ElixirOfGreaterAgility = { actionId: ActionId.fromItemId(9187), value: AgilityElixir.ElixirOfGreaterAgility, minLevel: 38 };
+export const ElixirOfLesserAgility = { actionId: ActionId.fromItemId(3390), value: AgilityElixir.ElixirOfLesserAgility, minLevel: 18 };
+export const ScrollOfAgility = { actionId: ActionId.fromItemId(10309), value: AgilityElixir.ScrollOfAgility };
+
+export const AGILITY_CONSUMES_CONFIG = [
+	{ config: ElixirOfTheMongoose, stats: [Stat.StatAgility] },
+	{ config: ElixirOfGreaterAgility, stats: [Stat.StatAgility] },
+	{ config: ElixirOfLesserAgility, stats: [Stat.StatAgility] },
+	{ config: ScrollOfAgility, stats: [Stat.StatAgility] },
+] as ConsumableStatOption<AgilityElixir>[];
+
+export const makeAgilityConsumeInput = makeConsumeInputFactory({consumesFieldName: 'agilityElixir'});
+
+// Strength
+export const JujuPower = { actionId: ActionId.fromItemId(12451), value: StrengthBuff.JujuPower, minLevel: 46 };
+export const ElixirOfGiants = { actionId: ActionId.fromItemId(9206), value: StrengthBuff.ElixirOfGiants, minLevel: 46 };
+export const ElixirOfOgresStrength = { actionId: ActionId.fromItemId(3391), value: StrengthBuff.ElixirOfOgresStrength, minLevel: 20 };
+export const ScrollOfStrength = { actionId: ActionId.fromItemId(10310), value: StrengthBuff.ScrollOfStrength };
+
+export const STRENGTH_CONSUMES_CONFIG = [
+	{ config: JujuPower, stats: [Stat.StatStrength] },
+	{ config: ElixirOfGiants, stats: [Stat.StatStrength] },
+	{ config: ElixirOfOgresStrength, stats: [Stat.StatStrength] },
+	{ config: ScrollOfStrength, stats: [Stat.StatStrength] },
+] as ConsumableStatOption<StrengthBuff>[];
+
+export const makeStrengthConsumeInput = makeConsumeInputFactory({consumesFieldName: 'strengthBuff'});
+
+// Other
+export const BoglingRootDebuff = makeBooleanConsumeInput({actionId: ActionId.fromItemId(5206), fieldName: 'boglingRoot'});
```

Comment:
Extremely minor, but should be `BoglingRootBuff` rather than `BoglingRootDebuff`